packages/realm/src/Realm.ts

```diff
@@ -822,7 +822,7 @@ export class Realm {
   /**
    * Deletes the provided Realm object, or each one inside the provided collection.
    */
-  delete(subject: RealmObject | RealmObject[] | List | Results): void {
+  delete<T = DefaultObject>(subject: RealmObject<T> | RealmObject<T>[] | List<T> | Results<T>): void {
```

Comment:
I am not sure if this is the right way to go about this, especially with regards to List/Results, will need more testing, but this is for the case where you pass a interface returned by `Realm.create<IDog>` and then try `realm.delete(dog)`

Reply:
I noticed you added `any` to remove before, should it be something like `RealmObject<any>` instead?

Reply:
I think this needs an overloaded definition to better depict these types, similar to `create` above.  I don't think `any` is the answer

Reply:
I guess this is also a question of should we be encouraging this? `IDog` as an interface isn't necessarily managed by Realm, it would probably be good to discourage it in terms of type-checking...